app/domain/authentication/common/validate_resource_restrictions.rb

```diff
@@ -0,0 +1,58 @@
+require 'command_class'
+
+module Authentication
+  module Common
+
+    ValidateResourceRestrictions = CommandClass.new(
+      dependencies: {
+        extract_resource_restrictions: ExtractResourceRestrictions.new,
+        logger:                        Rails.logger
+      },
+      inputs:   %i(authenticator_name service_id host_name account constraints authentication_request)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::ValidatingResourceRestrictions.new(@host_name))
+
+        resource_restrictions = extract_resource_restrictions
+        validate_resource_restrictions_configuration(resource_restrictions)
+        validate_request_matches_resource_restrictions(resource_restrictions)
+
+        @logger.debug(LogMessages::Authentication::ValidatedResourceRestrictions.new)
+      end
+
+      private
+
+      def extract_resource_restrictions
+        @extract_resource_restrictions.call(
+          authenticator_name: @authenticator_name,
+          service_id:         @service_id,
+          host_name:          @host_name,
+          account:            @account
+        )
+      end
```

Comment:
i would change this to:
```
def extract_resource_restrictions
	resource_restrictions
end

def resource_restrictions
  @resource_restrictions ||= @extract_resource_restrictions.call(
	  authenticator_name: @authenticator_name,
	  service_id:         @service_id,
	  host_name:          @host_name,
	  account:            @account
	)
end
```

this way you don't need a return object and can use `resource_restrictions` as a memoized object. it fits our style better.

Reply:
the `extract_resource_restrictions` method is used to tell the story in the `call` function and the `resource_restrictions` method is used for memoization

Reply:
Although I prefer the explicit way, I understand this style, and adhere to it.
Done.